client/src/types.ts

```diff
@@ -9,6 +9,12 @@ interface User {
   is_superuser: boolean
 }
 
+interface ResponseData {
+  detail: string,
+  errors: string[],
```

Comment:
I think this would be optional

Reply:
They are already optional in this interface definition. An object with only keys `detail, errors` is equally valid.